src/lib/storage/storage_manager.cpp

```diff
@@ -10,42 +11,48 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static auto instance = StorageManager{};
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  const bool insert_success = _tables.insert(std::make_pair(name, table)).second;
+  Assert(insert_success, name + " already exists, choose a different name!");
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  const auto values_erased = _tables.erase(name);
+  Assert(values_erased, name + " does not exist and cannot be deleted.");
 }
 
-std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  return nullptr;
-}
+std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const { return _tables.at(name); }
 
-bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  return false;
-}
+bool StorageManager::has_table(const std::string& name) const { return _tables.contains(name); }
 
 std::vector<std::string> StorageManager::table_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto table_names = std::vector<std::string>{};
+  table_names.reserve(_tables.size());
+
+  for (const auto& table : _tables) {
+    table_names.push_back(table.first);
+  }
+
+  // overhead added by the usage of an unordered_map, mainly needed to ensure an order for testing
```

Comment:
With `table_name` being used in `print`, it's technically not only used for testing. But the testing aspect probably also applies to the ordering of the tables in the print function, doesn't it?